File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    phone_book.sort()
    length = len(phone_book)
    for i in range(length - 1):
        check = len(phone_book[i])
        for j in range(i + 1, length):
            # if(phone_book[i][0] != phone_book[j][0]):
            #     break
            if(phone_book[j][: check] > phone_book[i]):
                break
            elif(phone_book[j][: check] == phone_book[i]):
                return False
    return True

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

class Solution {
    
    public Boolean solution(String[] phone_book) {
        Boolean answer = true;
        List<String> l = new ArrayList<>();
        for(String elem : phone_book) {
          l.add(elem);
        }
        l.sort(null);
        for(int i = 0; i < phone_book.length - 1; i++) {
          if(l.get(i + 1).startsWith(l.get(i))) {
            answer = false;
            break;
          }
        }
        return answer;
    }
}